✨ Feat : Project 생성 기능구현

seonghyun-1 opened the pull request:
- Valid 추가
- ValidException 추가

## PR 체크사항
PR이 다음 사항을 만족하는지 확인해주세요.

<!-- 
체크하려면 괄호 안에 "x"를 입력하세요. 
각 규칙은 Convention 문서에 있습니다.
PR 제목에 쓰는 prefix는 다음과 같습니다.
🚀 Release
🐛 Fix
✨ Feat
📝 Doc
♻️ Refactor
🔧 Chore
⏪️ Revert
🧪 Test
🎉 Init
-->

- [x] 커밋 제목 규칙
- [x] 커밋 메시지 작성 가이드라인
- [x] 라벨, 담당자, 리뷰어 지정


## PR 타입
어떤 유형의 PR인지 체크해주세요.

<!-- 체크하려면 괄호 안에 "x"를 입력하세요. -->

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Documentation content changes
- [ ] Other... Please describe:


## PR 설명
Project 생성기능추가 

## 작업사항
- Project 생성기능 추가
- Valid 추가
 - 제목 30자 제한
 - 요약 150자 제한
- ValidException 추가

seonghyun-1 committed:
[feat] Project 생성기능 추가수정 ( Valid 및 Exception 추가 )

- Valid 추가
- ValidException 추가

seonghyun-1 committed:
[feat] Project 생성기능 추가(수정)

- dto이름 수정 
 - ProjectRequestDto -> ProjectCreateRequestDto

seonghyun-1 committed:
[feat] 프로젝트 생성기능(수정)
- exception아래 mapper폴더가 들어가있던 오류 수정

hyunmuam commented on src/main/java/com/team1/TBBCloneCoding/project/controller/ProjectController.java:
```diff
@@ -0,0 +1,25 @@
+    @PostMapping("")
+    public ResponseEntity<ResponseDto> createProject(@RequestBody @Valid ProjectCreateRequestDto projectCreateRequestDto, @AuthenticationPrincipal UserDetailsImpl userDetails){
+        ResponseDto responseDto = projectService.createProject(projectCreateRequestDto, userDetails.getMember());
+        return new ResponseEntity<>(responseDto, HttpStatus.OK);
```
`return new ResponseEntity<>(responseDto, HttpStatus.OK);`에서 `<>`를 지워도 되지 않을까여?

seonghyun-1 replied:
수정하겠습니다!

hyunmuam reviewed (changes requested):
`Project` 엔티티에서 `ImageList`지워주세요

seonghyun-1 committed:
[fix] feature-#4 버그 픽스
- 버그 수정 및 주석 삭제

seonghyun-1 committed:
[fix] feature-#4 픽스
- 수정 및 주석 삭제

hyunmuam commented on src/main/java/com/team1/TBBCloneCoding/project/entity/Project.java:
```diff
@@ -8,6 +8,7 @@
 
 import javax.persistence.*;
 import java.time.LocalDateTime;
+import java.util.List;
```
사용하지 않는 import는 지워주셔도 될 것 같습니다.

seonghyun-1 replied:
확인했습니다!

hyunmuam reviewed (changes requested):
리뷰 확인 부탁드립니다.

seonghyun-1 committed:
[feat] 프로젝트 생성기능 수정
- 프로젝트 생성에서 이미지 업로드 시 이미지 엔티티 및 프로젝트 엔티티의 매핑기능추가

seonghyun-1 committed:
[fix] 프로젝트 생성기능 수정
- 안쓰는 라이브러리 import 제거
- return new ResponseEntity<>(responseDto, HttpStatus.OK);에서 <> 제거

hyunmuam commented on src/main/java/com/team1/TBBCloneCoding/project/service/ProjectService.java:
```diff
@@ -0,0 +1,41 @@
+        List<Long> imageNumberList = projectCreateRequestDto.getImageList();
+        Image image;
+        for(Long i : imageNumberList){
+            // 저장된 이미지를 레포지토리 가져와서 연결 -> 저장로직은 구현이 안됨
```
`-> 저장로직은 구현이 안됨` 은 지워주셔도 될 듯 합니다.

seonghyun-1 committed:
[fix] 프로젝트 생성기능구현(주석수정)

- 주석처리제거

seonghyun-1 committed:
[feat] 프로젝트 추가기능 (월) 회의 요구사항 반영
- ProjectCreateRequestDto에 thumbnailList, contentImageList로 변경(기존에는 imageList 1개 였음)
- image 엔티티 수정 -> whichContent 값 추가
 - 그안에 String으로 thumbnailImage, contentImage로 값 저장
 - Mapper 사용하지 않음( 추가 리팩토링 사항 )

seonghyun-1 committed:
[feat] 프로젝트 생성기능
- 프로젝트 엔티티에 content 추가

JeongO41 commented on src/main/java/com/team1/TBBCloneCoding/project/controller/ProjectController.java:
```diff
@@ -0,0 +1,25 @@
+
+@RestController
+@RequiredArgsConstructor
+@RequestMapping("/api/project")
```
"/api/tumblebug" 으로 수정해야 할 듯합니다!

JeongO41 reviewed (approved):
확인했습니다!
